### PR TITLE
pass correlation IDs

### PR DESCRIPTION
**Pull request by pdsarin**
not yet ready for review

### CODE REVIEW

File: mbq/client/client.py [hunk 1]
```diff
@@ -27,6 +28,14 @@ def _make_url(self, url):
     def _make_request(self, method, url, *args, **kwargs):
         if self._headers:
             kwargs['headers'] = dict(self._headers, **kwargs.get('headers', {}))
+        else:
+            kwargs['headers'] = {}
+
+        if 'correlation_id' in kwargs:
```
Comment (mcverry) [inline mbq/client/client.py]: Spitballing:

imagine the following workflow is using mbq.client for each hop
BFF -> CX -> Darkside -> Iris
this should generate (at least) 4 SNS workflow events (one from each service), each of which would have the same correlation id.

I am wondering if there is a way to instrument automatically storing received correlation-ids from the incoming request and then magically have detected and forwarded with outgoing requests?

File: setup.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import setuptools
 
 
-__version__ = '0.0.10'
+__version__ = '0.0.11'
```
Comment (mcverry) [inline setup.py]: heads up: iirc, you will need to tag the commit to push it to pypi

File: mbq/client/client.py (hunk 1 above)
Comment (pdsarin) [reply mbq/client/client.py]: wondering the same thing. Let's keep our eyes open for such an opportunity.

Comment (thieman) [reply mbq/client/client.py]: Might be misunderstanding Ryan's question but I've used thread locals and global middleware before to make this process relatively magical. Easy to consume those thread locals in logs, outbound requests, etc

Comment (pdsarin) [reply mbq/client/client.py]: agree. I think that is Ryan's question. `mbq.client` is agnostic of the web framework so I don't think that it's the right place to put that stuff.

**Pull request merged**